### PR TITLE
Fix the handling of memory allocation for tmpfs.

### PR DESCRIPTION
**Pull request by rhatdan**
If the total memory on the system is less then the max memory in cgroup
then do not set memory limit on tmpfs, allow the kernel to set the default.

### CODE REVIEW

**Comment by rhatdan** [comment]
@mrunalp PTAL

**Comment by rhatdan** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1406435 
Covers this issue in a little more depth.

**Comment by mrunalp** [comment]
Looks fine. Will test and merge.

**Comment by mrunalp** [comment]
This crashed when I tried to start docker container setting memory.

**Comment by mrunalp** [comment]
This is what I see in the logs:
```
19003aa33a5e56ad0c9bb35eafd1e7065b.scope/memory.limit_in_bytes
Dec 20 13:39:17 dhcp-16-119.sjc.redhat.com oci-systemd-hook[8522]: systemdhook <debug>: LIMIT: 1073741824
Dec 20 13:39:17 dhcp-16-119.sjc.redhat.com oci-systemd-hook[8522]: systemdhook <debug>: Limit in bytes: 1073741824
Dec 20 13:39:17 dhcp-16-119.sjc.redhat.com oci-systemd-hook[8522]: systemdhook <error>: Failed to mount tmpfs at /tmp: Invalid argument
```

**Comment by rhatdan** [comment]
Ok what command did you use to cause this?

File: src/systemdhook.c [hunk 1]
```diff
@@ -510,9 +527,9 @@ static int prestart(const char *rootfs,
 		}
 
 		if (!strcmp("", mount_label)) {
-			rc = asprintf(&options, "mode=1777,size=%" PRIu64 "k", memory_limit_in_kb);
+			rc = asprintf(&options, "mode=1777%s", memory_str);
```
Comment (mrunalp) [inline src/systemdhook.c]: Missing a comma between mode and size.

File: src/systemdhook.c [hunk 2]
```diff
@@ -487,15 +496,23 @@ static int prestart(const char *rootfs,
...
 
 	pr_pdebug("Limit in bytes: ""%" PRIu64 "\n", memory_limit_in_bytes);
 
-	/* Set it to half of limit in kb */
-	uint64_t memory_limit_in_kb = memory_limit_in_bytes / 2048;
+	total_memory = get_mem_total();
+	if (memory_limit_in_bytes < total_memory) {
+		/* Set it to half of limit in kb */
+		uint64_t memory_limit_in_kb = memory_limit_in_bytes / 2048;
+		snprintf(memory_str, sizeof(memory_str)-1 , "size=%" PRIu64 "k", memory_limit_in_kb);
```
Comment (mrunalp) [inline src/systemdhook.c]: I think you can add the comma before size here.

**Comment by mrunalp** [comment]
@rhatdan  
```
docker run -it --rm --memory=1G fedora-httpd
```

**Comment by rhatdan** [comment]
Put the "," back in, and am setting up a test.

**Comment by rhatdan** [comment]
docker run -ti --memory=1G fedora mount | grep /tmp
tmpfs on /tmp type tmpfs (rw,nosuid,nodev,relatime,context="system_u:object_r:container_file_t:s0:c133,c919",size=524288k)

docker run -ti fedora mount | grep /tmp
tmpfs on /tmp type tmpfs (rw,nosuid,nodev,relatime,context="system_u:object_r:container_file_t:s0:c197,c202")

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**